packages/dtslint-runner/src/main.ts

```diff
@@ -161,16 +161,16 @@ export async function runDTSLint({
   return allFailures.length;
 }
 
-function getExpectedFailures(onlyRunAffectedPackages: boolean) {
+function getExpectedFailures(onlyRunAffectedPackages: boolean, dependents: readonly string[]) {
   if (onlyRunAffectedPackages) {
     return undefined;
   }
 
   return new Set(
     (readFileSync(joinPaths(__dirname, "../expectedFailures.txt"), "utf8") as string)
       .split("\n")
-      .filter(Boolean)
       .map((s) => s.trim())
+      .filter(onlyRunAffectedPackages ? (line) => line && dependents.includes(line) : Boolean)
```

Comment:
This function returns early if `onlyRunAffectedPackages` is true, so I'm not sure that this is having the affect you intend in that case.

Reply:
Yeah, this PR is effectively a noop (besides running this after `trim`), as `onlyRunAffectedPackages ` is guaranteed to be false by this point, and therefore `Boolean` will be chosen.

Reply:
Ah, yeah, that early exit conditional was supposed to be removed.